packages/sol/src/verification.ts

```diff
@@ -0,0 +1,408 @@
+  MAX_MESSAGE_TEXT_LENGTH,
+  MAX_NOTE_TEXT_LENGTH,
+  MAX_PASSWORD_LENGTH,
+  MAX_PLAN_ALIAS_LENGTH,
+  MAX_PLAN_DISCOUNT,
+  MAX_PLAN_WEEKLY_MINUTES,
+  MAX_RATING_TEXT_LENGTH,
+  MAX_RATING_VALUE,
+  MAX_RULE_DURATION_MINUTES,
+  MAX_RULE_TITLE_LENGTH,
+  MAX_TUTOR_ABOUT_TEXT_LENGTH,
+  MAX_TUTOR_BIO_TEXT_LENGTH,
+  MAX_TUTOR_NOTICE_DURATION,
+  MAX_USER_AGE,
+  MIN_DISCOUNT_VALUE,
+  MIN_FEEDBACK_TEXT_LENGTH,
+  MIN_INTERVIEW_LEVEL,
+  MIN_MESSAGE_TEXT_LENGTH,
+  MIN_NOTE_TEXT_LENGTH,
+  MIN_PASSWORD_LENGTH,
+  MIN_PLAN_ALIAS_LENGTH,
+  MIN_PLAN_DISCOUNT,
+  MIN_PLAN_WEEKLY_MINUTES,
+  MIN_RATING_TEXT_LENGTH,
+  MIN_RATING_VALUE,
+  MIN_RULE_DURATION_MINUTES,
+  MIN_RULE_TITLE_LENGTH,
+  MIN_TUTOR_NOTICE_DURATION,
+  MIN_USER_AGE,
+  NUMBERS_ONLY_REGEX,
+  PASSWORD_REGEX,
+  PHONE_NUMBER_REGEX,
+  USER_NAME_REGEX,
+} from "@/constants";
+import { Bank, FieldError, WithdrawMethod } from "@litespace/types";
+import { getSafeInnerHtmlText } from "./utils";
+
+export function isValidEmail(email: string): FieldError.InvalidEmail | true {
+  if (!EMAIL_REGEX.test(email)) return FieldError.InvalidEmail;
+  return true;
+}
+
+export function isValidPassword(
+  password: string
+):
+  | FieldError.PasswordTooShort
+  | FieldError.PasswordTooLong
+  | FieldError.PasswordMissingLetters
+  | FieldError.PasswordMissingNumbers
+  | true {
+  if (!NUMBERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingNumbers;
+  if (!LETTERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingLetters;
+  if (password.length > MAX_PASSWORD_LENGTH) return FieldError.PasswordTooShort;
+  if (password.length < MIN_PASSWORD_LENGTH) return FieldError.PasswordTooShort;
+  if (PASSWORD_REGEX.test(password)) return true;
+  return true;
+}
+
+export function isValidUserName(
+  userName: string
+): FieldError.InvalidUserName | true {
+  if (!USER_NAME_REGEX.test(userName)) return FieldError.InvalidUserName;
+  return true;
+}
+
+export function isValidUserBirthYear(
+  userBirthYear: number
+): FieldError.UserTooOld | FieldError.UserTooYoung | true {
+  const currentYear = new Date().getFullYear();
+  const minYear = currentYear - MAX_USER_AGE;
+  const maxYear = currentYear - MIN_USER_AGE;
+
+  if (userBirthYear < minYear) return FieldError.UserTooOld;
+  if (userBirthYear > maxYear) return FieldError.UserTooYoung;
+  return true;
+}
+
+export function isValidTutorBio(
+  bio: string
+): FieldError.EmptyBio | FieldError.TooLongBio | true {
+  if (!bio.length) return FieldError.EmptyBio;
+  if (bio.length > MAX_TUTOR_BIO_TEXT_LENGTH) return FieldError.TooLongBio;
+
+  return true;
+}
+
+export function isValidTutorAbout(
+  about: string
+):
+  | FieldError.InvalidTutorAbout
+  | FieldError.EmptyTutorAbout
+  | FieldError.TooLongTutorAbout
+  | true {
+  const tutorAboutText = getSafeInnerHtmlText(about);
+
+  if (!tutorAboutText.length) return FieldError.EmptyTutorAbout;
+  if (!HTML_REGEX.test(about)) return FieldError.InvalidTutorAbout;
```

Comment:
The final result of `getSafeInnerHtmlText` is a plain text. This mean that this check is not necessary.